core/common/src/main/java/alluxio/file/PositionReadable.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.file;
+
+import com.google.common.base.Preconditions;
+import io.netty.buffer.ByteBuf;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+/**
+ * Position read interface. All implementer should be thread-safe.
+ */
+// TODO(lu) change to simple interface and have default implementation in abstract class
+// after block store is removed so that PagedFileReader does not need to extend abstract class
+// BlockReader
+public interface PositionReadable extends Closeable {
```

Comment:
This is the main interface.
Suppose to have the simple interface and have default implementation in an abstract class.
But current PagedFileReader needs to extend another abstract class and one class cannot extend two abstract classes. 
Will change it once we remove block store totally.